### PR TITLE
Generate Pokemon header files using json and inja

### PR DESCRIPTION
**Pull request by V-FEXrt**
The PR makes it so that many of the pokemon data header files are generated from json. This is a checkpoint towards a larger goal of allowing any given pokemon to be defined fully by a single json file. The end goal would be allowing the creation/deletion of a pokemon from the game engine by simply adding or removing a json file (scripts that depend on a given pokemon are unchanged). Doing this makes it easier to manipulate pokemon either by manually editing the json or by writing a tool that understands the json.

Pokemon that have game version variants (dugtrio and deoxys) have two files, one for each version, and are handled by adding a `"game_variant": "FIRERED"` key to the json.

**Major changes**
- json file for each pokemon is introduced under `data/pokemon`
- The following files are now generated at build time and removed from the repo 
  - src/data/pokemon/base_stats.h
  - src/data/pokemon/egg_moves.h
  - src/data/pokemon/evolution.h
  - src/data/pokemon/level_up_learnsets.h
  - src/data/pokemon/level_up_learnset_pointers.h
  - src/data/pokemon/tmhm_learnsets.h
  - src/data/pokemon/tutor_learnsets.h
- The build now depends on the [shell tool jq](https://stedolan.github.io/jq/)

### CODE REVIEW

File: json_data_rules.mk [hunk 1]
```diff
@@ -1,15 +1,77 @@
...
+
+$(DATA_C_SUBDIR)/pokemon/level_up_learnset_pointers.h: $(DATA_ASM_SUBDIR)/pokemon/pokemon_list.json $(DATA_ASM_SUBDIR)/pokemon/templates/level_up_learnset_pointers.h.inja
+	$(JSONPROC) $^ $@
+
+# Generate level_up_learnsets.h
+AUTO_GEN_TARGETS += $(DATA_C_SUBDIR)/pokemon/level_up_learnsets.h
+GENERATED_HEADERS += $(DATA_C_SUBDIR)/pokemon/level_up_learnsets.h
+
+$(DATA_C_SUBDIR)/pokemon/level_up_learnsets.h: $(DATA_ASM_SUBDIR)/pokemon/pokemon_list.json $(DATA_ASM_SUBDIR)/pokemon/templates/level_up_learnsets.h.inja
+	$(JSONPROC) $^ $@
```
Comment (PikalaxALT) [inline json_data_rules.mk:42]: These can probably be simplified.

```makefile
POKEMON_JSON_HEADERS := $(DATA_C_SUBDIR)/pokemon/level_up_learnset_pointers.h $(DATA_C_SUBDIR)/pokemon/level_up_learnsets.h ...

AUTO_GEN_TARGETS += $(POKEMON_JSON_HEADERS)
GENERATED_HEADERS += $(POKEMON_JSON_HEADERS)

$(POKEMON_JSON_HEADERS): $(DATA_C_SUBDIR)/%.h: $(DATA_ASM_SUBDIR)/pokemon/pokemon_list.json $(DATA_ASM_SUBDIR)/%.inja
    $(JSONPROC) $^ $@

**Review by PikalaxALT — CHANGES_REQUESTED** [review]

File: json_data_rules.mk (hunk 1 above)
Comment (V-FEXrt) [reply json_data_rules.mk]: Done!

**Review by PikalaxALT — APPROVED** [review]

**Comment by GriffinRichards** [comment]
I think my response to <https://github.com/pret/pokeemerald/pull/1123> applies similarly here in that I wonder if this is the kind of thing that's better left as a fork. I understand that this is meant to be a step towards making incrementally adding pokemon easier, but A. I don't think this even addresses the most annoying parts of that process, and B. a pokemon editor tool sounds like the kind of thing that's going to break easily, given the sorts of deviating changes people are likely to make. There's also some things that are going to be messy to extract. There's a bunch of miscellaneous tables indexed by species that people are going to need to update anyway, and people are likely to add more tables of their own.

**Comment by V-FEXrt** [comment]
Closing per discussion in discord.

**Pull request closed without merging**